psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1784,6 +1792,550 @@ PopulationAnalysisCalc::compute_lowdin_charges(bool print_output) {
+        }
+
+        // Update populations, widths, and densities
+        Nai = Nai_next;
+        Sai = Sai_next;
+        rho_0_points = rho_0_points_next;
+        rho_a_0_points = rho_a_0_points_next;
+
+        if (print_output && debug >= 1) outfile->Printf("   @MBIS iter %3d:  %.3e\n", iter, delta_rho_max_0);
+
+        if (delta_rho_max_0 < conv) {
+            if (print_output && debug >= 1) outfile->Printf("  MBIS Atomic Density Converged\n\n");
+            is_converged = true;
+            break;
+        }
+
+        iter += 1;
+    }
+
+    if (!is_converged) throw ConvergenceError<int>("MBIS", max_iter, conv, delta_rho_max_0, __FILE__, __LINE__);
+
+    // => Post-Processing <= //
+ 
+    // Atomic density, as defined in Equation 5 of Verstraelen et al. (Negative to account for negative charge of electron)
+    std::vector<double> rho_a(num_atoms * total_points, 0.0);
+#pragma omp parallel for
+    for (int atom = 0; atom < num_atoms; atom++) {
+        for (size_t point = 0; point < total_points; point++) {
+            rho_a[atom * total_points + point] = -rho[point] * rho_a_0_points[atom * total_points + point] / rho_0_points[point];
+        }
+    }
+
+    // Kronecker Delta
+    std::vector<std::vector<double>> k_delta = {{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}};
+
+    // Non-redundant cartesian quadrupole components
+    std::vector<std::vector<int>> qpole_inds = {{0, 0},
+                                                {0, 1},
+                                                {0, 2}, 
+                                                {1, 1}, 
+                                                {1, 2},
+                                                {2, 2}};
+
+    // Non-redundant cartesian octupole components
+    std::vector<std::vector<int>> opole_inds = {{0, 0, 0},
+                                                {0, 0, 1},
+                                                {0, 0, 2}, 
+                                                {0, 1, 1}, 
+                                                {0, 1, 2}, 
+                                                {0, 2, 2}, 
+                                                {1, 1, 1}, 
+                                                {1, 1, 2}, 
+                                                {1, 2, 2}, 
+                                                {2, 2, 2}};
+
+    // Non-redundant cartesian multipoles
+    auto mpole = std::make_shared<Matrix>("MBIS Charges: (a.u.)", num_atoms, 1);
+    auto dpole = std::make_shared<Matrix>("MBIS Dipoles: (a.u.)", num_atoms, 3);
+    auto qpole = std::make_shared<Matrix>("MBIS Quadrupoles: (a.u.)", num_atoms, 6);
+    auto opole = std::make_shared<Matrix>("MBIS Octupoles: (a.u.)", num_atoms, 10);
+    //auto qpole_stone = std::make_shared<Matrix>("MBIS Quadrupoles: (a.u.)", num_atoms, 6);
+    //auto opole_stone = std::make_shared<Matrix>("MBIS Octupoles: (a.u.)", num_atoms, 10);
```

Comment:
Were the `_stone` variants used for testing? I see them in the loop below. If they are correct and useful in the future a short comment about that and what they are would help. Otherwise better remove.

Reply:
The current multipoles are given by the convention in the Horton software. `_stone` represents multipoles as given by the convention in _The Theory of Intermolecular Forces, 2nd Edition_ by Anthony Stone.